festive-ferrets/backend/nchan/utils.py

```diff
@@ -0,0 +1,18 @@
+def shorten_text(text: str, length: int) -> str:
+    """
+    Shorten text to length chars cut to last space or new line and add '...'
+    :param text: Text to be shortened
+    :type text: str
+    :param length: Maximum text length after cut, not counting '...'
+    :type length: int
+    :return: Shortened text
+    :rtype: str
+    """
+    short_text = text
+    if len(short_text) > length:
+        short_text = short_text[:length + 1]
+        cut_index = max(short_text.rfind(' '), short_text.rfind('\n'))
+        if cut_index != -1:
+            short_text = short_text[:cut_index]
+        short_text += '...'
```

Comment:
It's a bit counter-intuitive to me that the string you get back could be longer than the length you've specified. A lot of similar functions take the length of the placeholder (in your case, the `"..."`) into account when "cutting" the text. I guess it's less critical here, but the reason is that you're sometimes dealing with character count restrictions and use a shorten function to ensure a certain maximum length.